### PR TITLE
Fix youtube error mapping codes to fit the Iglu schema (closes #1111)

### LINKED ISSUE #1111: Fix valid youtube error
**Describe**
When embedding a youtube video on a page we may get `MISSING_EMBED_PERMISSION` error, which is not in the enum type [here](https://github.com/snowplow/snowplow-javascript-tracker/blob/4dec01b9a314ca11d88372afa4b2dc67a431ee0d/plugins/browser-plugin-youtube-tracking/src/contexts.ts#L126). 
This results to the event being invalidated as well ([schema](https://github.com/snowplow/iglu-central/blob/cb00a49406207243ae22d282441359735f8850e6/schemas/com.youtube/youtube/jsonschema/1-0-0#L56)). As discussed with @greg-el , there seems to be a mismatch with the error codes mapping [here](https://github.com/snowplow/snowplow-javascript-tracker/blob/4dec01b9a314ca11d88372afa4b2dc67a431ee0d/plugins/browser-plugin-youtube-tracking/src/constants.ts#L45).

**To Reproduce**
Probably by embedding a youtube video without embed permissions.

**Expected behavior**
The event with the youtube error being valid.

**Additional context**
The corresponding youtube schema may need to be patched as well.

### PR DESCRIPTION
**Pull request by igneel64**
### Description

Fix Youtube error mapping codes to match the [YT error schema](https://github.com/snowplow/iglu-central/blob/cb00a49406207243ae22d282441359735f8850e6/schemas/com.youtube/youtube/jsonschema/1-0-0#L56).

### Note
Probably this implementation never worked, so we should keep in mind that clients updating to the new version, might see new events coming up in the good pipeline.

closes #1111

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://docs.snowplowanalytics.com/docs/contributing/contributor-license-agreement/ to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

**Review by greg-el — COMMENTED** [review]
LGTM!

I just thought I'd check before approving if this is supposed to be merged into a release branch? It's currently pointed at master.

**Comment by igneel64** [comment]
Yes on `release/3.6.1` or `release/3.7.0` which is yet to be seen based on the types of changes.

**Review by greg-el — APPROVED** [review]

**Pull request merged**
